File: Algorithm/python/BAEKJOON/CLASS 2/7568.py
# TRY 2
'''
- brute force(완전탐색)
'''
num_student = int(input())
student_list = []

for _ in range(num_student):
    weight, height = map(int, input().split())
    student_list.append((weight, height))

for i in student_list:
    rank = 1
    for j in student_list:
        if i[0] < j[0] and i[1] < j[1]:
            rank += 1
    print(rank, end = " ")

# # TRY 1
# '''
# - ASCII Code를 활용하여 대문자 알파벳을 이용하여 dict 생성
# '''
# N = int(input())
# group = dict()
# for i in range(N):
#     weight, height = map(int, input().split(" "))
#     grade = 1
#     information = dict()
#     information['weight'] = weight
#     information['height'] = height
#     information['grade'] = grade
#     group[chr(65+i)] = information
# for j in range(N-1):
#     if group[chr(65+j)]['weight'] < group[chr(65+j+1)]['weight'] and group[chr(65+j)]['height'] < group[chr(65+j+1)]['height']:
#         group[chr(65+j)]['grade'] += 1
#     else:
#         group[chr(65+j)]['grade'] = group[chr(65+j+1)]['grade']
# print(group)

File: Algorithm/javascript/BAEKJOON/CLASS 2/7568.js
// TRY 1

const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];
rl.on("line", (line) => {
  if (!line) {
    rl.close();
  } else {
    input.push(line);
  }
}).on("close", () => {
  solution(input);
  process.exit();
});

function solution(input) {
  let N = Number(input[0]);
  let group = [];
  for (let i = 1; i <= N; i++) {
    let [weight, height] = input[i].split(" ").map((e) => Number(e));
    group.push([weight, height]);
  }
  let rankArray = [];
  for (compare1 of group) {
    let rank = 1;
    for (compare2 of group) {
      if (compare1[0] < compare2[0] && compare1[1] < compare2[1]) {
        rank++;
      }
    }
    rankArray.push(rank);
  }
  console.log(rankArray.join(" "));
}
